.changeset/gorgeous-kiwis-drive.md

```diff
@@ -0,0 +1,13 @@
+---
+'xstate': major
+---
+
+This update restricts invoked `Subscribable`s to `EventObject`s,
+so that type inference can be done on which `Subscribable`s are
+allowed to be invoked.
+
+Existing `MachineConfig`s that invoke `Subscribable`s that are not
+`Subscribable<EventObject>`s should be updated accordingly:
+either their emissions are desired, and should thus be acted on
+as events, or they are not, in which case an action would
+probably be a better choice than a service.
```

Comment:
I don't quite understand this advice - could you rephrase this somehow or explain this in more detail in an answer to me?

Reply:
Sure, but it depends on whether you see this as a breaking change ore not. If current implementations follow the documentation, no machine configs will exist that have subscribable services that emit non-events. However, strictly following the current typings, it might be possible that such machines _do_ exist, even though they are quiete useless since the emitted values of those subscribables would not end up anywhere. Therefore my advice was to either emit events ifthe output of the subscribable is relevant, or use an action instead of a service if not. Not sure it is the perfect choice in every case though.